djangocassandra/db/backends/cassandra/compiler.py

```diff
@@ -52,41 +81,112 @@ def __init__(
             self.connection,
             self.query.model
         )
-        self.cql_query = self.column_family_class.objects.all().values_list(
-            *[field.column for field in self.meta.fields]
-        )
 
-    def _get_query_results(
-        self,
-        low_mark=None,
-        high_mark=None
-    ):
-        if None is not high_mark and high_mark <= low_mark:
-            return None
+        self.column_names = [
+            field.db_column if field.db_column else field.column
+            for field in fields
+        ]
+        self.indexed_columns = [
+            field.db_column if field.db_column else field.column
+            for field in fields if field.db_index
+        ]
+        self.cql_query = self.column_family_class.objects.values_list(
+            *self.column_names
+        ).all()
 
-        results = []
-        for result in self.cql_query:
-            result_dict = {}
+    def _get_rows_by_indexed_column(self, range_predicate):
+        if range_predicate._is_exact():
+            rows = self.cql_query.filter(**{
+                range_predicate.column: range_predicate.start
+            })
 
-            for i in xrange(len(self.meta.fields)):
-                result_dict[self.meta.fields[i].name] = result[i]
+        else:
+            if None is range_predicate.start:
+                start_op = None
 
-            results.append(result_dict)
+            else:
+                if range_predicate.start_inclusive:
+                    start_op = 'gte'
+                else:
+                    start_op = 'gt'
+                start_op = '__'.join([
+                    range_predicate.column,
+                    start_op
+                ])
 
-        return results
+            if None is range_predicate.end:
+                end_op = None
 
-    def fetch(
-        self,
-        low_mark=None,
-        high_mark=None
-    ):
-        results = self._get_query_results(
-            low_mark,
-            high_mark
+            else:
+                if range_predicate.end_inclusive:
+                    end_op = 'lte'
+                else:
+                    end_op = 'lt'
+                end_op = '__'.join([
+                    range_predicate.column,
+                    end_op
+                ])
+
+            query = self.cql_query
+            if None is not start_op:
+                query = query.filter(**{
+                    start_op: range_predicate.start
+                })
+    
+            if None is not end_op:
+                query = query.filter(**{
+                    end_op: range_predicate.end
+                })
+
+            rows = [query]
```

Comment:
Pretty sure this isn't what I meant, just want to return the query object I believe.